contracts/contracts/VerificationGateway.sol

```diff
@@ -266,6 +267,13 @@ contract VerificationGateway
         }
     }
 
+    /**
+    Declare that this is a verification gateway. See BLSWallet#setTrustedGateway.
+     */
+    function isVerificationGateway() external pure returns (bool) {
+        return true;
+    }
```

Comment:
I would recommend using the [EIP-165](https://eips.ethereum.org/EIPS/eip-165) spec for this sort of check. `OpenZepplin` should have all the needed contracts/utils at https://docs.openzeppelin.com/contracts/4.x/api/utils#IERC165 . This will also allow 3rd parties a standardized way to check if a contract is a verification gateway.

In the future, this could also be worth doing for `BLSWallet` as well.

Reply:
Hmm doing eip165 was the original plan actually. But we decided against it because the whole idea of changing gateways is that the gateway may work in a different way. There aren't any functions on a gateway which truly define what a gateway is, because the defining feature is that the wallet allows calls from the gateway.

Therefore, we wanted to be able to signal that the gateway is a gateway without putting any additional constraints on what a gateway needs to be.

Having said that, this is probably moot, see https://github.com/jzaki/bls-wallet/pull/114#issuecomment-1023738629.